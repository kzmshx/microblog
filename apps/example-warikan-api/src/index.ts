import { createApp } from "./app";
import { config } from "@repo/example-warikan-config";

let app;

if (process.env.DATA_PATH) {
  const data_path = process.env.DATA_PATH;
  app = createApp(`${data_path}/groups.json`, `${data_path}/expenses.json`);
} else {
  throw new Error();
}

app.listen(config.apiPort, () => {
  console.log(`Start on port ${config.apiPort}`);
});
