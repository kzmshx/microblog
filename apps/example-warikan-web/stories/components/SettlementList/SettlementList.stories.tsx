import { Meta, StoryObj } from "@storybook/react";
import SettlementList from "../../../src/components/SettlementList/SettlementList.tsx";
import { Settlement } from "../../../src/type.ts";

const meta: Meta<typeof SettlementList> = {
  title: "SettlementList",
  component: SettlementList,
};

export default meta;

type Story = StoryObj<typeof SettlementList>;

const settlements: Settlement[] = [
  { from: "Amy", to: "Bob", amount: 100 },
  { from: "Cathy", to: "Bob", amount: 200 },
];

export const Default: Story = {
  args: {
    settlements,
  },
};
