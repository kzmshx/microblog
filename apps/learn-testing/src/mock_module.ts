import fs from "fs";

export function readFile(path: string): string {
  return fs.readFileSync(path, { encoding: "utf-8" });
}
