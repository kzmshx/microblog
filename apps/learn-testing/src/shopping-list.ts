export class ShoppingList {
  private items: string[] = [];

  addItem(item: string): void {
    this.items.push(item);
  }

  removeItem(item: string): void {
    const index = this.items.indexOf(item);
    if (index === -1) {
      throw new Error(`Item ${item} not found`);
    }
    this.items.splice(index, 1);
  }

  getItems(): string[] {
    return this.items;
  }
}
