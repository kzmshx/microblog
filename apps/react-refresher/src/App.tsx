import PostList from "@/components/PostList";

export default function App() {
  return (
    <main>
      <PostList />
    </main>
  );
}
