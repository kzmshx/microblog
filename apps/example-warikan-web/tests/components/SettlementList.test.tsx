import { Settlement } from "../../src/type";
import SettlementList from "../../src/components/SettlementList/SettlementList";
import { render } from "@testing-library/react";
import { expect } from "vitest";

describe("SettlementList", () => {
  test("snapshot renders", () => {
    // Arrange
    const settlements: Settlement[] = [{ from: "A", to: "B", amount: 10 }];
    // Act
    const { container } = render(<SettlementList settlements={settlements} />);
    // Assert
    expect(container).toMatchSnapshot();
  });
});
