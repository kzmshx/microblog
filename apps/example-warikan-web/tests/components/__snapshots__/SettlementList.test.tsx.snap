// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettlementList > snapshot renders 1`] = `
<div>
  <div
    class="chakra-stack css-1anxrbq"
  >
    <p
      class="chakra-text css-x39wmu"
    >
      清算方法
    </p>
    <ul
      class="css-1s6xsev"
      role="list"
    >
      <li
        class="css-1v5z18m"
      >
        <div
          class="chakra-stack css-bot922"
        >
          <div
            class="css-0"
          >
            A
             → 
            B
          </div>
          <div
            class="css-0"
          >
            10
            円
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
